tools/link.py

```diff
@@ -166,18 +166,17 @@ def setup_environment_settings():
   # Note: we need to actually modify ENVIRONMENTS variable here before the parsing,
   # because some JS code reads it back so modifying parsed info alone is not sufficient.
   if settings.SHARED_MEMORY and settings.ENVIRONMENT:
-    settings.ENVIRONMENT += ',worker'
+    settings.ENVIRONMENT.append('worker')
 
   # Environment setting based on user input
-  environments = settings.ENVIRONMENT.split(',')
-  if any(x for x in environments if x not in VALID_ENVIRONMENTS):
+  if any(x for x in settings.ENVIRONMENT if x not in VALID_ENVIRONMENTS):
```

Comment:
I wonder if we can make `ENVIRONMENT` a `set()` instead of a `list()`. Then you could just do `if settings.ENVIRONMENT - VALID_ENVIRONMENTS:`.

Reply:
There is some precedent for using `Set` for settings on the JS side for settings in python today are only strings, numbers or lists.

This change seems simpler as is, we can followup up if we decide it makes a difference.

Reply:
Yeah just took a moment to read, and thought at least `if set(settings.ENVIRONMENT) - VALID_ENVIRONMENTS:` would be simple enough if you don't want to change the setting itself.

But fine either way.